pages/mesosphere/dcos/services/data-science-engine/1.0.1/security/index.md

```diff
@@ -0,0 +1,264 @@
+      "secret_envkeys": "PLACEHOLDER,SECRET_ENVVAR"
+    }
+  }
+}
+```
+<p class="message--note"><strong>NOTE: </strong> It is mandatory to provide values for <tt>service_account</tt> and <tt>service_account_secret</tt> in the service configuration in order to access any secrets.</p>
+
+This configuration places the contents of `{{ model.packageName }}/my-secret-file` into the `target-secret-file` as well as the `PLACEHOLDER` environment variable. Additionally, the contents of `{{ model.packageName }}/my-secret-envvar` are exported to the `SECRET_ENVVAR` and written to the `placeholder-file`.
+
+<p class="message--note"><strong>NOTE: </strong> If the content size of binary secrets is greater than 4KB, Mesos' security module will reject container execution due to the overhead.</p>
+
+# Authenticating to your {{ model.techName }} instance
+
+You can run multiple installations of {{ model.techName }} by changing the `service.name` option during installation.
+Each instance can have different authentication mechanisms configured.
+
+## Password Authentication (default)
+
+The default {{ model.techName }} password is set to`jupyter`. You can override it with the `service.jupyter_password` option.
+
+## OpenID Connect
+
+You can choose to enable OpenID Connect authentication. The OpenID Connect flow will be triggered if `oidc.enabled` is
+`true` and both `oidc.discovery_uri` and `oidc.client_secret` are set, since they are the minimal options.
+
+The default client name is `notebook` and it can be overridden with the `oidc.client_id` option.
+
+Here is an example of a simple OpenID Connect configuration for {{ model.techName }}:
+
+```json
+{
+  "service": {
+      "service_account": "<service-account-id>",
+      "service_account_secret": "<service-account-secret>",
+  },
+  "oidc": {
+      "enabled": true,
+      "discovery_uri": "https://keycloak.example.com/auth/realms/notebook/.well-known/openid-configuration",
+      "client_id": "{{ model.serviceName }}-client",
+      "client_secret": "11111111-2222-3333-4444-555555555555"
+  }
+}
+```
+
+<p class="message--note"><strong>NOTE: </strong> It is mandatory to provide values for <tt>service_account</tt> and <tt>service_account_secret</tt> in the service configuration in order to access any secrets.</p>
+
+<!-- There are a few more options for advanced OpenID Connect configuration, that can be found in the `Oidc` section when
+installing {{ model.techName }} from the catalog in the DC/OS UI.  -->
+
+
+# Using Arbitrary Secrets
+
+Suppose you want to use credentials for private repositories. First of all, create a credential property file with following content:
+
+```properties
+user = username
+pwd = password
+host = repository.com
+```
+
+You can use this credential file in any notebook by using `%AddDeps` as follows:
+
+```notebook
+%AddDeps org.name artifact version --credential /path/to/credentials.prop --repository https://repository.com/artifactory/example-repo-local/
+```
```

Comment:
let's add a sentence that in order to see all the configuration parameters of `%AddDeps` magic, a user should run `%AddDeps` without any arguments or flags

Reply:
Is the use of the word "magic" standard in the industry?

Reply:
Yes, it's a Jupyter Notebook terminology, "magic" means something similar to macros executed in a notebook cell.

Reply:
Done